server/src/main/java/org/elasticsearch/search/aggregations/bucket/nested/NestedAggregator.java

```diff
@@ -193,11 +194,14 @@ void processBufferedChildBuckets() throws IOException {
 
             for (; childDocId < currentParentDoc; childDocId = childDocs.nextDoc()) {
                 cachedScorer.doc = childDocId;
-                final long[] buffer = bucketBuffer.buffer;
-                final int size = bucketBuffer.size();
-                for (int i = 0; i < size; i++) {
-                    collectBucket(sub, childDocId, buffer[i]);
-                }
+                final int docId = childDocId;
+                bucketBuffer.stream().mapToLong(Long::longValue).forEach(bucket -> {
```

Comment:
Maybe use a regular for loop here? I notice that we have been avoiding to use `steam()` in hot code paths. I think we should do this here too? But Maybe I'm wrong here.

Reply:
I had done the stream because I was trying to avoid materializing a primitive array. But I realized now I could just iterate over the boxed values and let java do the auto unboxing on the call to collectBucket.